src/mdoc.targets

```diff
@@ -0,0 +1,269 @@
+      <MDocVersion Condition=" '%(Identity)' == 'mdoc' ">@(PackageReference -> '%(Version)')</MDocVersion>
+      <MDocPackagePath Condition=" '%(Name)' == 'mdoc' ">@(PackageDefinitions -> '%(ResolvedPath)')</MDocPackagePath>
+      <MDocPackagePath Condition=" '$(MDocPackagePath)' == '' ">$(NuGetPackageRoot)mdoc\$(MDocVersion)</MDocPackagePath>
+      <MDocToolPath>$(MDocPackagePath)\tools\mdoc.exe</MDocToolPath>
+      <MDocOutputName>$(TargetName).xml</MDocOutputName>
+      <MDocOutputPath>$(TargetDir)$(MDocOutputName)</MDocOutputPath>
+      <_ShouldGenerateDocs Condition=" '$(MDocDocumentationDirectory)' != '' and Exists('$(MDocDocumentationDirectory)\index.xml') ">true</_ShouldGenerateDocs>
+    </PropertyGroup>
+  </Target>
+
+  <!--
+    ============================================================
+    MDocGenerateDocs
+
+    Generated the XML documentation file using mdoc.
+    This runs during the build phase.
+    ============================================================
+  -->
+  <Target Name="MDocGenerateDocs"
+          AfterTargets="CoreCompile" DependsOnTargets="_MDocSetup">
+    <ItemGroup Condition=" '$(_ShouldGenerateDocs)' == 'true' ">
+      <FileWrites Include="$(MDocOutputPath)" />
+    </ItemGroup>
+    <Exec Condition=" '$(_ShouldGenerateDocs)' == 'true' "
+          Command="$(_ManagedExeLauncher) &quot;$(MDocToolPath)&quot; export-msxdoc --debug --out=&quot;$(MDocOutputPath)&quot; &quot;$(MDocDocumentationDirectory)&quot;" />
+  </Target>
+
+  <!--
+    ============================================================
+    MDocAddFilesToPackage
+
+    Make sure the mdoc output goes into the final package.
+    This runs during the package phase.
+    ============================================================
+  -->
+  <Target Name="MDocAddFilesToPackage"
+          BeforeTargets="_GetBuildOutputFilesWithTfm" DependsOnTargets="_MDocSetup">
+    <ItemGroup Condition=" '$(_ShouldGenerateDocs)' == 'true' ">
+      <BuildOutputInPackage Include="$(MDocOutputPath)" TargetFramework="$(TargetFramework)" TargetPath="$(MDocOutputName)" />
+    </ItemGroup>
+  </Target>
+
+  <!--
+    ============================================================
+    MDocUpdateDocs
+
+    Update the docs in the documentation directory using the
+    compiled assemblies.
+    This runs as a separate task.
+    ============================================================
+  -->
+  <Target Name="MDocUpdateDocs"
+          DependsOnTargets="_MDocSetup">
+    <!-- bits for resolving references on Windows -->
+    <PropertyGroup Condition=" '$(OS)' == 'Windows_NT' ">
+      <FrameworkReferenceAssemblyPath>$(VsInstallRoot)\Common7\IDE\ReferenceAssemblies\Microsoft\Framework</FrameworkReferenceAssemblyPath>
+      <WindowsKitsReferenceAssemblyPath>$(MSBuildProgramFiles32)\Windows Kits\10\References\10.0.17763.0</WindowsKitsReferenceAssemblyPath>
```

Comment:
I believe that our WindowsSDK version is different.

Reply:
I took it from here: https://github.com/xamarin/XamarinCommunityToolkit/blob/main/src/CommunityToolkit/Xamarin.CommunityToolkit/Xamarin.CommunityToolkit.csproj#L5